node/src/consensus/aura_consensus.rs

```diff
@@ -179,30 +196,36 @@ impl ConsensusMechanism for AuraConsensus {
         task_manager: &mut TaskManager,
         client: Arc<FullClient>,
         triggered: Option<Arc<std::sync::atomic::AtomicBool>>,
+        sync_service: Arc<SyncingService<Block>>,
     ) -> Result<(), sc_service::Error> {
         let client_clone = client.clone();
         let triggered_clone = triggered.clone();
         let slot_duration = self.slot_duration(&client)?;
         task_manager.spawn_essential_handle().spawn(
-        "babe-switch",
-        None,
-        Box::pin(async move {
-            let client = client_clone;
-            let triggered = triggered_clone;
-            loop {
-                // Check if the runtime is Babe once per block.
-                if let Ok(c) = sc_consensus_babe::configuration(&*client) {
-                    if !c.authorities.is_empty() {
-                        log::info!("Babe runtime detected! Intentionally failing the essential handle `babe-switch` to trigger switch to Babe service.");
-						if let Some(triggered) = triggered {
-                        	triggered.store(true, std::sync::atomic::Ordering::SeqCst);
-						};
-                        break;
-                    }
-                };
-                tokio::time::sleep(slot_duration.as_duration()).await;
-            }
-        }));
+            "babe-switch",
+            None,
+            Box::pin(async move {
+                let client = client_clone;
+                let triggered = triggered_clone;
+                loop {
+                    // Check if the runtime is Babe once per block.
+                    if let Ok(c) = sc_consensus_babe::configuration(&*client) {
+                        // Aura Consensus uses the hybrid import queue which is able to import both
+                        // Aura and Babe blocks. Wait until sync finishes before switching to the
+                        // Babe service to not break warp sync.
+                        let syncing = sync_service.status().await.is_ok_and(|status| status.warp_sync.is_some() || status.state_sync.is_some());
+                        if !c.authorities.is_empty() && !syncing {
+                            log::info!("Babe runtime detected! Intentionally failing the essential handle `babe-switch` to trigger switch to Babe service.");
+                            if let Some(triggered) = triggered {
+                                triggered.store(true, std::sync::atomic::Ordering::SeqCst);
+                            };
+                            break;
+                        }
+                    };
+                    tokio::time::sleep(slot_duration.as_duration()).await;
```

Comment:
What happens if the actual block duration exceeds the slot duration? And in general, if the trigger activates in unsynchronized way after the block production?